source/fuzz/transformation_inline_function.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 André Perez Maselco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_TRANSFORMATION_INLINE_FUNCTION_H_
+#define SOURCE_FUZZ_TRANSFORMATION_INLINE_FUNCTION_H_
+
+#include "source/fuzz/protobufs/spirvfuzz_protobufs.h"
+#include "source/fuzz/transformation.h"
+#include "source/fuzz/transformation_context.h"
+#include "source/opt/ir_context.h"
+
+namespace spvtools {
+namespace fuzz {
+
+class TransformationInlineFunction : public Transformation {
+ public:
+  explicit TransformationInlineFunction(
+      const protobufs::TransformationInlineFunction& message);
+
+  TransformationInlineFunction(
+      uint32_t function_call_id,
+      const std::map<uint32_t, uint32_t>& result_id_map);
+
+  // - |message_.result_id_map| must map the instructions of the called function
+  // to fresh ids.
+  // - |message_.function_call_id| must be an OpFunctionCall instruction. The
+  // called function must not have an early return.
+  bool IsApplicable(
```

Comment:
Similarly, fix indentation here.

Also add a note that the called function must not use OpUnreachable or OpKill, and briefly say that this is to guard against making hte module invalid when the caller is inside a continue construct.

With respect to this new bullet that I'm proposing: can you open an issue to note that we could make the spirv-fuzz inliner more aggressive by allowing functions that use OpKill / OpUnreachable to be inlined if the function call is *not* part of a continue construct, and do a TODO(issue URL) comment here?